pkg/cmd/release-payload-controller/legacy_job_status_controller.go

```diff
@@ -0,0 +1,286 @@
+// the CIConfigurationName of the job.  This is our only way of mapping the individual
+// results back to their respective CIConfiguration.
+//
+// The LegacyJobStatusController operates only on the ReleasePayloads that specify:
+//   - .spec.payloadVerificationConfig.legacyResults = True
+//
+// and writes to the following locations:
+//   - .status.blockingJobResults[].results
+//   - .status.informingJobResults[].results
+//
+// The legacy results are very much married to the ReleaseConfig that's defined on the
+// "Source" imagestream when the respective release is created.  Processing of these
+// results are only concerned with the Blocking and Informing jobs defined in the
+// ReleaseConfig.  Therefore, this controller will not process anything related to the
+// automatic upgrade jobs produced by the release-controller.
+type LegacyJobStatusController struct {
+	*ReleasePayloadController
+
+	imageStreamLister imagev1lister.ImageStreamLister
+}
+
+func NewLegacyJobStatusController(
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	imageStreamInformer imagev1informer.ImageStreamInformer,
+	eventRecorder events.Recorder,
+) (*LegacyJobStatusController, error) {
+	c := &LegacyJobStatusController{
+		ReleasePayloadController: NewReleasePayloadController("Legacy Job Status Controller",
+			releasePayloadInformer,
+			releasePayloadClient,
+			eventRecorder.WithComponentSuffix("legacy-job-status-controller"),
+			workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "LegacyJobStatusController")),
+		imageStreamLister: imageStreamInformer.Lister(),
+	}
+
+	c.syncFn = c.sync
+
+	releasePayloadInformer.Informer().AddEventHandler(&cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.Enqueue,
+		UpdateFunc: func(old, new interface{}) { c.Enqueue(new) },
+		DeleteFunc: c.Enqueue,
+	})
+
+	return c, nil
+}
+
+func (c *LegacyJobStatusController) sync(ctx context.Context, key string) error {
+	klog.V(4).Infof("Starting LegacyJobStatusController sync")
+	defer klog.V(4).Infof("LegacyJobStatusController sync done")
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	klog.V(4).Infof("Processing ReleasePayload: '%s/%s' from workQueue", namespace, name)
+
+	// Get the ReleasePayload resource with this namespace/name
+	originalReleasePayload, err := c.releasePayloadLister.ReleasePayloads(namespace).Get(name)
+	// The ReleasePayload resource may no longer exist, in which case we stop processing.
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// If .spec.payloadVerificationConfig.legacyResults in not set, then we don't have anything to do...
+	if !originalReleasePayload.Spec.PayloadVerificationConfig.LegacyResults {
```

Comment:
you need every other controller to have the "do nothing if legacy results" logic, right?

Reply:
The Prowjob Controller is the only controller that should need a block to prevent it from wasting any time on these payloads (which shouldn't have any corresponding ProwJobs left in the system due to their age anyway).
The other controllers all should react accordingly.